Added getEdgeIntegralOfPolygonalChain to tensor mesh.

Pbellive opened the pull request:
Hi Lars,
              I've added a docstring and tests for this function. The main use is in computing the source term in an electromagnetic problem where the source is defined by a polygonal chain describing the wire path.

Pbellive committed:
Added getEdgeIntegralOfPolygonalChain to tensor mesh. Computes integral of unit current defined by polygonal chain projected onto edges.

lruthotto commented on src/Mesh/getEdgeIntegralOfPolygonalChain.jl:
```diff
@@ -0,0 +1,234 @@
+
+function getStraightLineCurrentIntegral(hx,hy,hz,ax,ay,az,bx,by,bz)
+
+# function [sx,sy,sz] = getStraightLineCurrentIntegral(hx,hy,hz,ax,ay,az,bx,by,bz)
```
can you make this a docstring as well, while you're at it?

lruthotto commented on src/Mesh/getEdgeIntegralOfPolygonalChain.jl:
```diff
@@ -0,0 +1,234 @@
+export getEdgeIntegralOfPolygonalChain
+
+using Base.BLAS
```
where do you use BLAS here?

lruthotto commented on src/Mesh/getEdgeIntegralOfPolygonalChain.jl:
```diff
@@ -0,0 +1,234 @@
+  ix = findlast((ax .>= x[1:nx-1]) & (ax .<= x[2:nx]))
+  iy = findlast((ay .>= y[1:ny-1]) & (ay .<= y[2:ny]))
+  iz = findlast((az .>= z[1:nz-1]) & (az .<= z[2:nz]))
+  msg = @sprintf("(%d,%d,%d) \n", ix, iy, iz)
```
shall we make the prints optional?

lruthotto commented on src/Mesh/getEdgeIntegralOfPolygonalChain.jl:
```diff
@@ -0,0 +1,234 @@
+          and phix_i, phiy_i, phiz_i = 0 at all other edges.
+        
+        INPUT
+        mesh ...... OcTree mesh
```
type of mesh does not fit function

lruthotto commented on src/Mesh/getEdgeIntegralOfPolygonalChain.jl:
```diff
@@ -0,0 +1,234 @@
+        polygon[1,:] == polygon[end,:]
+"""
+
+function getEdgeIntegralOfPolygonalChain(M::TensorMesh3D,polygon::Array{Float64,2};
```
Do you think this method can be extended for other meshes (i.e. include RegularMesh or OcTree Mesh)?

lruthotto commented on src/Mesh/getEdgeIntegralOfPolygonalChain.jl:
```diff
@@ -0,0 +1,234 @@
+px, py, pz = polygon[:,1], polygon[:,2], polygon[:,3]
+
+# nodal grid
+x  = [x0; x0 + cumsum(hx)]
```
you may replace this using getNodalAxes(mesh)

lruthotto commented:
Thanks Patrick. All looks good, but I've made a few comments in the code.

Pbellive commented:
Hi Lars,
              I've addressed all your comments in the code. Blas calls were from a previous version of the code. I've removed the `using Base.BLAS`. We have a version of this code for OcTree meshes (written by Christoph). This code has been adapted from that. I can write a version for regular meshes as well but I'm not sure I'll have time in the next couple of days. I'll make a PR when I get it done.

lruthotto commented:
great job, Patrick. Thanks a lot!
